snuba/consumers/consumer_builder.py

```diff
@@ -271,4 +274,7 @@ def build_base_consumer(self) -> StreamProcessor[KafkaPayload]:
         """
         Builds the consumer.
         """
-        return self.__build_consumer(self.__build_streaming_strategy_factory())
+        processor = self.__build_consumer(self.__build_streaming_strategy_factory())
+        self.commit_log_producer.close()
+        self.replacements_producer.close()
```

Comment:
You can't close the producers immediately after building the consumer. Otherwise they won't be able to do any producing anymore. Only close them when the consumer is shutting down.